### PR TITLE
Store panels and docks instances in their classes

### PR DESCRIPTION
**Pull request by trollodel**
Based on the work in #47520

This is a cleanup PR that reduces a bit the editor code dependencies to EditorNode.
It moves all the panel and dock instances, currently stored in EditorNode in their classes as a singleton.

This is one approach (IMO the simplest). The other approach is to store the node instances in a dedicated class called EditorSingletons or something like that (closer to what I did in #47520).

### CODE REVIEW

**Comment by KoBeWi** [comment]
Another thing that would be nice to do is unifying the usage of `EditorNode` singleton. Many plugins will have an `editor` property. EditorNode is passed in the constructor and saved to that property, but then it's only used half of the time, because the class sometimes still accesses the editor via singleton. I'd remove all these "cached" editors and just always access it using singleton. This way these classes won't need the EditorNode to be passed with constructor.

**Comment by trollodel** [comment]
@KoBeWi definitely agree. In fact, this is only the first of some editor related cleanup PRs I want to do, and the EditorNode parameter thing is something I already noted. I waited to do more work because I was unsure if this is desired.

**Comment by KoBeWi** [comment]
> I waited to do more work because I was unsure if this is desired.

tbh, it's better if you waited a bit longer, so the PR doesn't end like the previous one .-. I'll add a PR meeting label.

File: editor/node_dock.cpp
Comment (trollodel) [reply editor/node_dock.cpp:139]: Now that I checked more carefully, this case is the only one when I need to add a destructor and I add it into the cpp like the other destructors I modified to be consistent. However, I don't mind moving all the destructor I changed to the header.

**Comment by akien-mga** [comment]
We discussed this today in a PR review meeting and agreed with the change.

> In fact, this is only the first of some editor related cleanup PRs I want to do, and the EditorNode parameter thing is something I already noted. I waited to do more work because I was unsure if this is desired.

We also intend to cleanup `EditorNode` at some point, but we decided to postpone that work to after 4.0 is released because we don't have much bandwidth for this in the core team and a lot of pending 4.0 items to finish. So you're welcome to work on this, but be aware that most contributors likely won't have much bandwidth to review it for several months - so you might also prefer to wait for a more opportune time to participate in and/or lead discussions with core contributors and then get the agreed-upon changes done.

**Review by akien-mga — APPROVED** [review]
Should probably be rebased, then it should be good to merge if it's ready.

**Comment by KoBeWi** [comment]
The `editor` changes I mentioned could be done now, no? It doesn't require EditorNode modifications.

**Comment by trollodel** [comment]
> We also intend to cleanup EditorNode at some point, but we decided to postpone that work to after 4.0 is released because we don't have much bandwidth for this in the core team and a lot of pending 4.0 items to finish. So you're welcome to work on this, but be aware that most contributors likely won't have much bandwidth to review it for several months - so you might also prefer to wait for a more opportune time to participate in and/or lead discussions with core contributors and then get the agreed-upon changes done.

I didn't want to do massive, risky cleanup PRs for several reasons, including this one. So, I think it's better to start with low risks PRs like this one and the future editor parameter PR (so @KoBeWi is happy 😂).
In any case, feel free to ping me on RC when the topic is raised.

**Comment by akien-mga** [comment]
> The `editor` changes I mentioned could be done now, no? It doesn't require EditorNode modifications.

Yeah for sure. As long as there are people available to review low-impact and consensual changes, that's definitely fine.

**Comment by akien-mga** [comment]
Thanks!

File: editor/plugins/animation_player_editor_plugin.cpp
Comment (TokageItLab) [inline editor/plugins/animation_player_editor_plugin.cpp:1797]: It cause bug, since `editor->get_inspector()` and `editor->get_inspector_dock()` is different.

Comment (trollodel) [reply editor/plugins/animation_player_editor_plugin.cpp:1797]: I probably missed it in the rebase. Thanks for the fix.

**Pull request merged**
